scim/data_service_principal.go

```diff
@@ -0,0 +1,52 @@
+package scim
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// DataSourceUser returns information about user specified by user name
+func DataSourceServicePrincipal() *schema.Resource {
+	return &schema.Resource{
```

Comment:
please rewrite with the new data resource framework: https://github.com/databrickslabs/terraform-provider-databricks/blob/master/jobs/data_jobs.go